src/mono/mono/mini/mini-arm64.c

```diff
@@ -2887,7 +2887,8 @@ mono_arch_allocate_vars (MonoCompile *cfg)
 			offset += size;
 
 			cfg->arch.swift_error_var = ins;
-			cfg->used_int_regs |= 1 << ARMREG_R21;
+			if (cfg->method->wrapper_type == MONO_WRAPPER_MANAGED_TO_NATIVE)
+ 				cfg->used_int_regs |= 1 << ARMREG_R21;
```

Comment:
It might be helpful to explain here that in the NATIVE_TO_MANAGED case, the error register is functioning as an extra return register, and thus isn't treated as callee-save.